lib/drivers/medtronicDriver.js

```diff
@@ -975,87 +1044,113 @@ module.exports = function (config) {
-                                             tzoUtil: cfg.tzoUtil,
-                                             builder: cfg.builder});
-
-        // sort by log index
-        postrecords = _.sortBy(postrecords, function(d) { return d.index; });
-        // sort by time
-        postrecords = _.sortBy(postrecords, function(d) { return d.time; });
-
-        try {
-          for (var j = 0; j < postrecords.length; ++j) {
-            var datum = postrecords[j];
-            switch (datum.type) {
-              case 'basal':
-                simulator.basal(datum);
-                break;
-              case 'bolus':
-                simulator.bolus(datum);
-                break;
-              case 'wizard':
-                simulator.wizard(datum);
-                break;
-              case 'smbg':
-                simulator.smbg(datum);
-                break;
-              case 'pumpSettings':
-                simulator.pumpSettings(datum);
-                break;
-              case 'deviceEvent':
-                if (datum.subType === 'status') {
-                  simulator.suspendResume(datum);
-                } else if (datum.subType === 'alarm') {
-                  simulator.alarm(datum);
-                } else if (datum.subType === 'prime') {
-                  simulator.prime(datum);
-                } else if (datum.subType === 'reservoirChange') {
-                  simulator.rewind(datum);
-                } else if (datum.subType === 'timeChange') {
-                  simulator.changeDeviceTime(datum);
-                }
-                break;
-              default:
-                debug('[Hand-off to simulator] Unhandled type!', datum.type);
+          if(cgmErr) {
+            return cb(cgmErr,data);
+          }
+
+          var dateRecords =  _.filter(records, function(event) {return event.jsDate;}); // filter out objects without dates
+          if(dateRecords.length === 0) {
+            return cb(new Error('No records found on pump'), null);
+          }
+          var mostRecent = sundial.applyTimezone(dateRecords[dateRecords.length-1].jsDate, cfg.timezone).toISOString();
+          proc.init(cfg, data.settings);
+          console.log('Pump time changes:');
+          var result = proc.buildTimeChangeRecords(records, mostRecent, 'pump');
+          var postrecords = result.postrecords;
+          cfg.tzoUtil = result.tzoUtil;
+
+          var cgmDateRecords =  _.filter(cgmRecords, function(event) {return event.jsDate;}); // filter out objects without dates
+          if(cgmDateRecords.length === 0) {
+            console.log('No CBG records found on pump');
+          } else {
+            var cgmMostRecent = sundial.applyTimezone(cgmDateRecords[cgmDateRecords.length-1].jsDate, cfg.timezone).toISOString();
+            console.log('CGM time changes:');
+            var cgmResult = proc.buildTimeChangeRecords(cgmRecords, cgmMostRecent, 'cgm');
+            // only pump time change events are uploaded
```

Comment:
This comment meant that only *pump* time change events are upload, not *CGM* time change events, as I expected them to be the same. After your comment on the Trello card I now realize that it's possible they may be different, so I'm now changing it to upload both.